kolibri/plugins/coach/assets/src/views/lessons/LessonSummaryPage/index.vue

```diff
@@ -0,0 +1,483 @@
+      resourceId => !this.removals.includes(resourceId)
+    );
+    this.removals = [];
+    this.setWorkingResources(remainingResources);
+    this.autoSave();
+  }, removalSnackbarTime);
+
+  export default {
+    name: 'lessonSummaryPage',
+    components: {
+      dropdownMenu,
+      ManageLessonModals,
+      StatusIcon,
+      contentIcon,
+      kButton,
+      uiIconButton,
+      kRouterLink,
+      InfoIcon,
+      coreTable,
+      progressBar,
+    },
+    data() {
+      return {
+        currentAction: null,
+        removals: [],
+      };
+    },
+    computed: {
+      lessonOptions() {
+        return map(this.actionsToLabelMap, (label, action) => ({
+          label: this.$tr(label),
+          action,
+        }));
+      },
+      actionsToLabelMap() {
+        return {
+          [LessonActions.COPY]: 'copyLesson',
+          [LessonActions.DELETE]: 'deleteLesson',
+          [LessonActions.EDIT_DETAILS]: 'editLessonDetails',
+        };
+      },
+      LessonActions() {
+        return LessonActions;
+      },
+      lessonSelectionRootPage() {
+        return selectionRootLink({ lessonId: this.lessonId, classId: this.classId });
+      },
+      removalMessage() {
+        const numberOfRemovals = this.removals.length;
+
+        if (!numberOfRemovals) {
+          return '';
+        } else if (numberOfRemovals === 1) {
+          return this.$tr('singleResourceRemovalConfirmationMessage', {
+            resourceTitle: this.resourceTitle(this.removals[0]),
+          });
+        }
+
+        return this.$tr('multipleResourceRemovalsConfirmationMessage', {
+          numberOfRemovals,
+        });
+      },
+    },
+    methods: {
+      handleSelectOption({ action }) {
+        this.currentAction = action;
+      },
+      lessonIsAssignedToClass(assignments) {
+        return (
+          assignments.length === 1 && assignments[0].collection_kind === CollectionTypes.CLASSROOM
+        );
+      },
+      getGroupName(assignment) {
+        return this.learnerGroups.find(lg => lg.id === assignment.collection).name;
+      },
+      resourceTitle(resourceId) {
+        return this.resourceContentNodes[resourceId].title;
+      },
+      resourceProgress(resourceId) {
+        return this.resourceContentNodes[resourceId].progress;
+      },
+      stageRemoval(index, resourceId) {
+        this.removals.push(resourceId);
+
+        this.createSnackbar({
+          text: this.removalMessage,
+          duration: removalSnackbarTime,
+          autoDismiss: true,
+          actionText: this.$tr('undoActionPrompt'),
+          actionCallback: () => {
+            this.removals = [];
+            this.clearSnackbar();
+          },
+        });
+
+        // cancel any pending calls to reset timer
+        this.cancelCommitRemovals();
+        this.commitRemovals();
+      },
+      commitRemovals,
+      cancelCommitRemovals: commitRemovals.cancel,
+      autoSave: debounce(function() {
```

Comment:
debouncing autosave to reduce servercalls. set to 6 secs right now. This means snackbar notifications are optimistic.

Reply:
Still a bit dubious about autoSave - what is the purpose of it, except to not have a save button?

Reply:
If it's already implemented, let's be careful not to change the plan too late in the game - I think we're about to move on to the Learner view

Reply:
Sure, I would urge to increase the interval though - 6 seconds seems too low!

Reply:
@khang thoughts on the Auto save?

Reply:
I had forgotten about the dirty checking in the resource layer that makes this repeated saving not an issue, and we should look at the autosave behaviour after implementation is finished to look for potential edge cases around synchronous editing/viewing. No need to change anything now.

Reply:
maybe 8 seconds

Reply:
Disregard my comment. Time does not matter, because it doesn't save unless there is a change anyway.

Reply:
@rtibbles yeah, resource layer helps when a delete/undo happens - we still send a request, but the resource layer understands that nothing was modified and doesn't send anything to the server. 

reorders, on the other hand, do.